mso/resource_ndo_schema_template_deploy.go

```diff
@@ -81,12 +81,43 @@ func resourceNDOSchemaTemplateDeployExecute(d *schema.ResourceData, m interface{
 		log.Printf("[DEBUG] MakeRestRequest failed with err: %s.", err)
 		return err
 	}
-	_, resp, err := msoClient.Do(req)
-	if err != nil || resp.StatusCode != 202 {
+	cont, resp, err := msoClient.Do(req)
+	if resp.StatusCode != 202 || err != nil {
 		log.Printf("[DEBUG] Request failed with resp: %v. Err: %s.", resp, err)
 		return err
 	}
 
+	taskId := cont.S("id").Data()
+	req, err = msoClient.MakeRestRequest("GET", fmt.Sprintf("%s/%s", path, taskId.(string)), nil, true)
+	if err != nil {
+		log.Printf("[DEBUG] MakeRestRequest failed with err: %s.", err)
+		return err
+	}
+
+	cont, resp, err = msoClient.DoWithRetryFunc(req, isTaskStatusPending)
+	if err != nil {
+		log.Printf("[DEBUG] Request failed with resp: %v. Err: %s.", resp, err)
+		return err
+	}
+
+	taskStatusContainer := cont.Search("operDetails", "taskStatus")
+	if taskStatusContainer != nil {
+		if status, ok := taskStatusContainer.Data().(string); ok {
+			if status == "Error" {
```

Comment:
could we combine to avoid a layer of nesting?
```suggestion
		if status, ok := taskStatusContainer.Data().(string); ok && status == "Error" {
```